### PR TITLE
Apic Client Fix

### PR DESCRIPTION
**Pull request by sayalinaval**
Change-Id: Ia017bfdae7397a89b1295877d59188532700c20a

### CODE REVIEW

**Review by irathore — APPROVED** [review]

**Comment by tbachman** [comment]
Please squash the commits.

File: aim/agent/aid/universes/aci/tenant.py [hunk 1]
```diff
@@ -302,18 +302,19 @@ def reset_ownership(self):
...
         super(AciTenantManager, self).__init__(*args, **kwargs)
         LOG.info("Init manager for tenant %s" % tenant_name)
         self.get_resources = get_resources
         self.apic_config = apic_config
         # Each tenant has its own sessions
-        self.aci_session = apic_session
         self.dn_manager = apic_client.DNManager()
         self.tenant_name = tenant_name
-        children_mos = get_children_mos(self.aci_session, self.tenant_name)
+        ws_context.establish_aci_session()
```
Comment (tbachman) [inline aim/agent/aid/universes/aci/tenant.py:315]: Is this needed? Isn't the session established already in the WebSocketContext class?

File: aim/agent/aid/universes/aci/tenant.py [hunk 2]
```diff
@@ -346,8 +347,9 @@ def noop(*args, **kwargs):
         self.error_handler = error.APICAPIErrorHandler()
         # For testing purposes
         self.num_loop_runs = float('inf')
-        self.ownership_mgr = OwnershipManager(apic_session, apic_config,
-                                              aim_system_id)
+        self.ws_context.establish_aci_session()
```
Comment (tbachman) [inline aim/agent/aid/universes/aci/tenant.py:350]: Same comment as above.

**Review by tbachman — CHANGES_REQUESTED** [review]
Overall, a really clean patch. Just a couple of comments.

File: aim/agent/aid/universes/aci/tenant.py [hunk 3]
```diff
@@ -128,7 +128,7 @@ class Root(acitoolkit.BaseACIObject):
     def __init__(self, *args, **kwargs):
         self.filtered_children = kwargs.pop('filtered_children', [])
         rn = kwargs.pop('rn')
-        ws_subscription_to = kwargs.pop('ws_subscription_to')
+        ac_subscription_to = kwargs.pop('ac_subscription_to')
```
Comment (tbachman) [inline aim/agent/aid/universes/aci/tenant.py:131]: I think you can leave this one as ws_subscription_to - this applies only to the websocket client, right?

File: aim/agent/aid/universes/aci/tenant.py [hunk 4]
```diff
@@ -302,26 +302,26 @@ def reset_ownership(self):
...
         self.apic_config = apic_config
         # Each tenant has its own sessions
-        self.aci_session = apic_session
         self.dn_manager = apic_client.DNManager()
         self.tenant_name = tenant_name
-        children_mos = get_children_mos(self.aci_session, self.tenant_name)
-        ws_subscription_to = self.apic_config.get_option(
+        children_mos = get_children_mos(ac_context.aci_session,
+                                        self.tenant_name)
+        ac_subscription_to = self.apic_config.get_option(
```
Comment (tbachman) [inline aim/agent/aid/universes/aci/tenant.py:317]: same comment here

**Review by tbachman — CHANGES_REQUESTED** [review]
Minor nit/change.

**Comment by tbachman** [comment]
Please back out OpenSSL change and squash commits, then rebase.

**Pull request merged**